### PR TITLE
[SPARK-8220][SQL]Add positive identify function

### PR DESCRIPTION
**Pull request by zhichao-li**
@chenghao-intel @adrian-wang

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -633,6 +634,15 @@ object SimplifyCasts extends Rule[LogicalPlan] {
 }
 
 /**
+ * Removes [[UnaryPositive]] identify function
+ */
+object RemovePositive extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
+    case UnaryPositive(data) => data
```
Comment (chenghao-intel) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: Change `data` => `child`?

File: sql/core/src/test/scala/org/apache/spark/sql/MathExpressionsSuite.scala [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 package org.apache.spark.sql
 
+import org.apache.spark.sql.execution.ExplainCommand
```
Comment (chenghao-intel) [inline sql/core/src/test/scala/org/apache/spark/sql/MathExpressionsSuite.scala]: Unused import?

File: sql/core/src/test/scala/org/apache/spark/sql/MathExpressionsSuite.scala [hunk 3]
```diff
@@ -262,4 +263,10 @@ class MathExpressionsSuite extends QueryTest {
       ctx.sql("SELECT negative(1), negative(0), negative(-1)"),
       Row(-1, 0, 1))
   }
+
+  test("positive") {
+    checkAnswer(
+      ctx.sql("""SELECT positive(1), positive(-1), positive("abc")"""),
```
Comment (chenghao-intel) [inline sql/core/src/test/scala/org/apache/spark/sql/MathExpressionsSuite.scala]: Use `selectExpr` instead, we'd better keep the same pattern with the other test case in the same file.

Comment (zhichao-li) [inline sql/core/src/test/scala/org/apache/spark/sql/MathExpressionsSuite.scala]: Just clone from the `negative` test, would change that later.

**Comment by chenghao-intel** [comment]
LGTM, can you remove the [WIP]? @rxin, can you review this one?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala [hunk 4]
```diff
@@ -58,6 +58,15 @@ case class UnaryMinus(child: Expression) extends UnaryArithmetic {
   protected override def evalInternal(evalE: Any) = numeric.negate(evalE)
 }
 
+case class UnaryPositive(child: Expression) extends UnaryArithmetic {
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala]: I think you can just name this Positive

**Comment by rxin** [comment]
actually i'm just going to merge this. UnaryPositiv vs Positive isn't that big of a deal here.

**Pull request closed without merging**